R/seroprevalence_data.R

```diff
@@ -243,52 +239,46 @@ get_sim_n_seropositive <- function(
 
 #' Function that generates a simulated serosurvey according to the specified FoI
 #'
-#' @param foi Numeric atomic vector corresponding to the desired
-#'   Force-of-Infection ordered from past to present
-#' @param sample_size_by_age Size of each age group specified by either an
-#'   atomic vector of the same size as `foi` or an integer. This corresponds to
-#'   the number of trials `size` in [rbinom][stats::rbinom].
-#' @param tsur Year in which the serosurvey was conducted.
-#' @param birth_year_min Minimum age of year in the simulated serosurvey.
+#' @inheritParams get_sim_n_seropositive
 #' @param survey_label Label for the resulting simulated serosurvey.
-#' @param antibody Type of antibody detected by the serosurvey
-#' @param test Type of test taken
-#' @param seed The seed for random number generation.
-#' @return Dataframe object containing the simulated data generated from \code{foi}
+#' @return Dataframe containing the simulated serosurvey.
 #' @examples
-#' \dontrun{
-#' sample_size_by_age <- 5
-#' foi <- rep(0.02, 50)
+#' n_years <- 50
+#' sim_data <- data.frame(
+#'   age_mean_f = seq(1,n_years),
+#'   tsur = 2050
+#' )
+#' foi <- rep(0.02, n_years)
+#' sample_size_by_age <- as.integer(runif(n = n_years, 5, 10))
 #' sim_data <- generate_sim_data(
+#'   sim_data = sim_data,
 #'   foi = foi,
 #'   sample_size_by_age = sample_size_by_age,
-#'   tsur = 2050,
-#'   birth_year_min = 2000,
 #'   survey_label = "sim_constant_foi"
 #' )
-#' }
 #' @export
-generate_sim_data <- function(foi,
+generate_sim_data <- function(sim_data,
+                              foi,
                               sample_size_by_age,
-                              tsur,
-                              birth_year_min,
                               survey_label,
-                              test = "fake",
-                              antibody = "IgG",
                               seed = 1234
-                              ){
-    sim_data <- data.frame(birth_year = c(birth_year_min:(tsur - 1))) %>%
-        mutate(tsur = tsur,
-            country = 'None',
-            test = test,
-            antibody = antibody,
-            survey = survey_label,
-            age_mean_f = tsur - .data$birth_year)
-    sim_n_seropositive <- get_sim_n_seropositive(sim_data, foi, sample_size_by_age, seed = seed)
-    sim_data <- sim_data %>%
-        mutate(counts = sim_n_seropositive$n_seropositive,
-               total = sample_size_by_age) %>%
-        prepare_serodata()
+) {
+  sim_data <- sim_data %>%
+    mutate(
+      country = "None",
+      test = "test",
+      antibody = "antibody",
```

Comment:
Do we need to set these?

Reply:
Not really. These variables are optional and are only used in some specific instances (e.g. `extract_seromodel_summary`) just to show additional information about the serosurveys. Removing them will trigger a warning in the tests once the simulated data is passed to `prepare_serodata()` and assign the value `"None"` to them (see #144 ), so I think is okay to remove them to simplify the simulation functions as much as possible.

Reply:
Addressed by [92f47b8](https://github.com/epiverse-trace/serofoi/pull/140/commits/92f47b82bd95e42b3abd03cbbfff7e111f2383d6)